internal/server/data/provider_test.go

```diff
@@ -42,7 +41,7 @@ func TestCreateProviderDuplicate(t *testing.T) {
 		createProviders(t, db, providerDevelop, providerProduction)
 
 		err := CreateProvider(db, &providerDevelop)
-		assert.ErrorIs(t, err, internal.ErrDuplicate)
+		assert.ErrorContains(t, err, "value for name already exists in providers")
```

Comment:
`err` is no longer a value so `assert.ErrorIs` is not appropriate. Looking through the doc, I did find `assert.ErrorType` but I don't know if its any better than `assert.ErrorContains`

Reply:
Definitely agree that our tests should be checking the type of the error, since that's an important part of the interface we are exposing from the `data` package.

To work with typed errors like this we'll need to use `errors.As`. I'd do something like this:

```go
var uerr UniqueConstraintError
assert.Assert(t, err, errors.As(err, &uerr), "error is wrong type %T", err)
expected := UniqueConstraintError{Column: "name", Table: "providers"}
assert.DeepEqual(t, uerr, expected)
```